ballerina/channel_csv_io.bal

```diff
@@ -135,3 +136,77 @@ isolated function getWritableCSVChannel(WritableChannel writableChannel) returns
     }
     return writableCSVChannel;
 }
+
+isolated function readHeadersFromCsvFile(string path) returns string[]|Error {
+    stream<string[], Error?>|Error csvContent = fileReadCsvAsStream(path);
+    if csvContent is Error {
+        if csvContent is FileNotFoundError {
+            return [];
+        } else {
+            return error GenericError("Error while reading the headers from the CSV file. " + csvContent.message());
+        }
+    } else {
+        do {
+            var csvLine = check csvContent.next();
+            return csvLine == () ? [] : csvLine["value"];
+        } on fail Error err {
+            check csvContent.close();
+            return error GenericError("Error while reading the headers from the CSV file. " + err.message());
+        }
+    }
+}
+
+isolated function validateCsvHeaders(string[] headersFromCSV, string[] headers) returns string[]|Error {
+    if headersFromCSV == [""] {
+        return [];
+    } else if headersFromCSV.length() == 0 {
+        return [];
```

Comment:
Do we need to validate these? I think this is already handled below.

Reply:
No if we don't handle these in the read, these two are not failure cases. first one is appending to non existent file and second one is appending to empty file. so these instances had to be handled separately from the checks below.

Reply:
Shall we merge these into one and add a comment here